crates/biome_css_semantic/src/semantic_model/specificity.rs

```diff
@@ -0,0 +1,180 @@
+                .into_iter()
+                .map(|s| s.map_or(ZERO_SPECIFICITY, |s| evaluate_any_compound_selector(&s)))
+                .reduce(|acc, e| acc.max(e))
+                .unwrap_or(ZERO_SPECIFICITY);
+
+            CLASS_SPECIFICITY + list_max
+        }
+        AnyCssPseudoClass::CssPseudoClassFunctionIdentifier(_) => CLASS_SPECIFICITY,
+        AnyCssPseudoClass::CssPseudoClassFunctionNth(_) => CLASS_SPECIFICITY,
+        AnyCssPseudoClass::CssPseudoClassFunctionRelativeSelectorList(selector_list) => {
+            if let Some(base) = selector_list
+                .name_token()
+                .ok()
+                .and_then(|name| evaluate_pseudo_function_selector(name.text()))
+            {
+                let list_max = selector_list
+                    .relative_selectors()
+                    .into_iter()
+                    .map(|relative_selector| {
+                        relative_selector
+                            .map_or(ZERO_SPECIFICITY, |s| evaluate_any_relative_selector(&s))
+                    })
+                    .reduce(|acc, e| acc.max(e))
+                    .unwrap_or(ZERO_SPECIFICITY);
+                base + list_max
+            } else {
+                ZERO_SPECIFICITY
+            }
+        }
+        AnyCssPseudoClass::CssPseudoClassFunctionSelector(s) => {
+            if let Some(base) = s
+                .name()
+                .ok()
+                .and_then(|name| evaluate_pseudo_function_selector(name.text()))
+            {
+                base + s.selector().map_or(ZERO_SPECIFICITY, |selector| {
+                    evaluate_any_selector(&selector)
+                })
+            } else {
+                ZERO_SPECIFICITY
+            }
+        }
+        AnyCssPseudoClass::CssPseudoClassFunctionSelectorList(selector_list) => {
+            if let Some(base) = selector_list
+                .name()
+                .ok()
+                .and_then(|name| evaluate_pseudo_function_selector(name.text()))
+            {
+                let list_max = selector_list
+                    .selectors()
+                    .into_iter()
+                    .map(|selector| {
+                        selector.map_or(ZERO_SPECIFICITY, |s| evaluate_any_selector(&s))
+                    })
+                    .reduce(|acc, e| acc.max(e))
+                    .unwrap_or(ZERO_SPECIFICITY);
+                base + list_max
+            } else {
+                ZERO_SPECIFICITY
+            }
+        }
+        AnyCssPseudoClass::CssPseudoClassFunctionValueList(_) => CLASS_SPECIFICITY,
+        AnyCssPseudoClass::CssPseudoClassIdentifier(_) => CLASS_SPECIFICITY,
+    }
+}
+
+fn evaluate_pseudo_selector(selector: &CssPseudoClassSelector) -> Specificity {
+    match selector.class() {
+        Ok(any_pseudo_class) => evaluate_any_pseudo_class(&any_pseudo_class),
+        Err(_) => ZERO_SPECIFICITY,
+    }
+}
+
+fn evaluate_any_subselector(selector: &AnyCssSubSelector) -> Specificity {
```

Comment:
Let's document this logic, maybe with a link to the spec

Reply:
After reviewing the spec, I noticed that specificity of `CssAttributeSelector` is incorrect.
Therefore, I have corrected this and added a link to the spec